things/service.go

```diff
@@ -181,7 +181,7 @@ func (svc service) ListClients(ctx context.Context, token, reqUserID string, pm
 
 	pm.IDs = ids
 
-	tp, err := svc.clients.RetrieveAllByIDs(ctx, pm)
+	tp, err := svc.SearchThings(ctx, token, pm)
```

Comment:
Why here we are calling `SearchThing` function
And not calling repo function `svc.clients.SearchBasicInfo(ctx, pm)`
`

Reply:
In the SearchThings were are checking if the user has authority to view the thing so it is separated from `listclients` when retrieving the basic info

Reply:
Authorization are done on above switch case.